CMakeLists.txt

```diff
@@ -140,7 +191,7 @@ file(GLOB COMPATSRC CONFIGURE_DEPENDS src/compat/compat.c)
 ############################################################################
 # libnotcurses-core (core shared library, core static library)
 file(GLOB NCCORESRCS CONFIGURE_DEPENDS src/lib/*.c src/lib/*.cpp)
-add_library(notcurses-core SHARED ${NCCORESRCS} ${COMPATSRC})
+add_library(notcurses-core SHARED ${NCCORESRCS} ${COMPATSRC} ${MACOS_COMPAT_SRC})
```

Comment:
i'd rather just use the existing compat.c and #ifdef within there, rather than need this everywhere.